### PR TITLE
DOCSP-43070: Delete documents

### PR DESCRIPTION
**Pull request by norareidy**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-43070
Staging - https://deploy-preview-53--docs-c.netlify.app/write/delete/

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [ ] Are all the links working?
- [ ] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/write/delete.txt [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+- ``mongoc_collection_delete_one()``, which deletes *the first document* that matches
+  the search criteria
+- ``mongoc_collection_delete_many()``, which deletes *all documents* that match the
+  search criteria
+
+Each delete function accepts the following parameters:
+
+- Collection: Specifies the collection to update.
+
+- Query filter document: Specifies which collection documents to update. For
```
Comment (stephmarie17) [inline source/write/delete.txt:52]: [s] Might be clearer to specify delete: 
```suggestion
- Query filter document: Specifies which collection documents to delete. For
```

File: source/write/delete.txt [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+You can perform delete operations by using the following functions:
+
+- ``mongoc_collection_delete_one()``, which deletes *the first document* that matches
+  the search criteria
+- ``mongoc_collection_delete_many()``, which deletes *all documents* that match the
+  search criteria
+
+Each delete function accepts the following parameters:
+
+- Collection: Specifies the collection to update.
```
Comment (stephmarie17) [inline source/write/delete.txt:50]: [s] I think the parameter names in this list can be emphasized per the style guide: 
```suggestion
- **Collection**: Specifies the collection to update.
```

**Review by stephmarie17 — APPROVED** [review]
Nice work ✨ a few suggestions in the comments but otherwise LGTM

File: source/write/delete.txt (hunk 1 above)
Comment (norareidy) [reply source/write/delete.txt:52]: oops thanks for catching!

File: source/includes/write/delete.c [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    {
+        // Deletes matching documents and attaches a comment to the operation
+        // start-delete-options
+        bson_t *filter = BCON_NEW ("name", "{", "$regex", BCON_UTF8 ("Mongo"), "}");
+        bson_error_t error;
+        bson_t opts;
+
+        bson_init(&opts);
```
Comment (joshbsiegel) [inline source/includes/write/delete.c:50]:
```suggestion

        bson_t opts;
        bson_init(&opts);
```
Suggest for readability. Not purposely trying to be a nuisance, just for consistency with the other docs I've seen :)

**Review by joshbsiegel — APPROVED** [review]
LGTM

**Pull request merged**